pkg/types/installconfig.go

```diff
@@ -218,6 +225,12 @@ type Networking struct {
 	DeprecatedClusterNetworks []ClusterNetworkEntry `json:"clusterNetworks,omitempty"`
 }
 
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
```

Comment:
Is this struct type (rather than just having `MachineNetworks []ipnet.IPNet` above) just for the possibility of future extensibility? (In the `ClusterNetworks` case we have a struct because we actually need multiple fields per sub-network.)

Reply:
Yes, and in general from an API perspective we'd say "always do this, never do single type arrays unless you have perfect future vision or your type is a lower level type that is already a list of type X"